source/usage-examples/replaceOne.txt

```diff
@@ -0,0 +1,110 @@
+==================
+Replace a Document
+==================
+
+.. default-domain:: mongodb
+
+You can replace a single document using the ``replaceOne()`` method on
+a ``MongoCollection`` object. This method removes all the existing fields
+and values from a document (except the ``_id`` field) and substitutes it
+with your replacement document.
+
+The ``replaceOne()`` method accepts a query filter that matches the
+document you want to replace and a replacement document that contains the
+data you want to save in place of the matched document. The ``replaceOne()``
+method only updates the first document that matches the filter.
+
+You can optionally pass an instance of ``ReplaceOptions`` to the method in
+order to specify the method behavior. For example, if you set the ``upsert``
+field of the ``ReplaceOptions`` object to ``true``, the operation inserts
+a new document from the fields in the replacement document if no documents
+match the query filter. See the link to the ``ReplaceOptions`` API
+documentation at the bottom of this page for more information.
+
+Upon successful execution, the ``replaceOne()`` method returns an instance
+of ``UpdateResult``.  You can retrieve information such as the number of
+documents modified calling the ``getModifiedCount()`` method. You can also
+retrieve the value of the document's ``_id`` field by calling the
+``getUpsertedId()`` method if you set ``upsert(true)`` in the
+``UpdateOptions`` instance and the operation resulted in an upsert.
```

Comment:
```suggestion
``UpdateOptions`` instance and the operation resulted in the insertion of a new document.
```

I would say that the operation is an upsert no matter what, but can result in either an update of an existing document or an insert of a new document.